Jenkinsfile-insertions

```diff
@@ -1,75 +1,75 @@
-							sh "gzip -f ${env.RELEASE_CURRENT}_${env.RELEASE_NUMBER}_before_addlinks.dump"
+							def release_current_before_addlinks_dump = "${env.RELEASE_CURRENT}_${env.RELEASE_NUMBER}_before_addlinks.dump"
+							sh "mysqldump -u$user -p$pass ${env.RELEASE_CURRENT} > $release_current_before_addlinks_dump"
+							sh "gzip -f $release_current_before_addlinks_dump"
 						}
 					}
 				}
 			}
 		}
+		// This stage builds the jar file using maven.
 		stage('Setup: Build jar file'){
-            		steps{
-		        	script{
-                    			dir('add-links'){
+  		steps{
+  			script{
+        	dir('add-links'){
 						sh "mvn clean package -DskipTests"
-                    			}
-          	    		}
-            		}
-        	}
+          }
+      	}
+      }
+    }
+		// This stage runs the insertion component of AddLinks. It will build links to various external resources and connect them to the relevant instances.
+		// It also connects these linkouts to ReferenceDatabase objects that pertain to their original resource.
 		stage('Main: AddLinks Insertions'){
 			steps{
 				script{
 					dir('add-links'){
-						withCredentials([file(credentialsId: 'Config', variable: 'FILE')]){
-							sh "cp $FILE src/main/resources/auth.properties"
+						withCredentials([file(credentialsId: 'Config', variable: 'ConfigFile')]){
+							sh "cp $ConfigFile src/main/resources/auth.properties"
 							sh "ln -sf src/main/resources resources"
 							sh "rm -f src/main/resources/db.properties"
-					    		sh """java -cp "\$(pwd)/resources" -Dconfig.location=\$(pwd)/resources/addlinks.properties -Dlog4j.configurationFile=\$(pwd)/resources/log4j2.xml -Xmx${env.JAVA_MEM_MAX}m -jar target/AddLinks-${env.ADDLINKS_VERSION}.jar file://\$(pwd)/resources/create-references-context.xml"""
+					    sh """java -cp "\$(pwd)/resources" -Dconfig.location=\$(pwd)/resources/addlinks.properties -Dlog4j.configurationFile=\$(pwd)/resources/log4j2.xml -Xmx${env.JAVA_MEM_MAX}m -jar target/AddLinks-${env.ADDLINKS_VERSION}.jar file://\$(pwd)/resources/create-references-context.xml"""
 							sh "rm -f src/main/resources/auth.properties"
 						}
 					}
 				}
 			}
 		}
-		stage('Post: Backup DB'){
+		// This stage backs up the release_current database after it is modified.
+		stage('Post: Backup release_current'){
 			steps{
 				script{
 					dir('add-links'){
 						withCredentials([usernamePassword(credentialsId: 'mySQLUsernamePassword', passwordVariable: 'pass', usernameVariable: 'user')]){
-							sh "mysqldump -u$user -p$pass ${env.RELEASE_CURRENT} > ${env.RELEASE_CURRENT}_${env.RELEASE_NUMBER}_after_addlinks.dump"
-							sh "gzip ${env.RELEASE_CURRENT}_${env.RELEASE_NUMBER}_after_addlinks.dump"
+							def release_current_after_addlinks_dump = "${env.RELEASE_CURRENT}_${env.RELEASE_NUMBER}_after_addlinks.dump"
+							sh "mysqldump -u$user -p$pass ${env.RELEASE_CURRENT} > $release_current_after_addlinks_dump"
+							sh "gzip $release_current_after_addlinks_dump"
```

Comment:
A project level utility function to backup and gzip databases would be good similar to https://github.com/reactome/add-links/pull/127/files/a6d3d0c59207884075bf057f022b07af5b784552..3809d4507f00851f8b0ec095e39200fccd4c4d54#r333680454